### DISCUSSION TITLE
How solve show dialog feature

### DISCUSSION THREAD

**Discussion by Kotlinlife** [arkivanov/Decompose#31] [Q&A] [answered]
How solve show dialog in decompose? 
Is it just differen child component?
thanks

**Comment by arkivanov** [comment, marked as answer]
Hello. I believe it depends on the complexity of the dialog. Relatively simple dialogs can be just part of the state of a component. E.g. there may be a field in the state, which is observed in the UI and the dialog is displayed.

Complex dialogs can be extracted to a separate component. Such a component can be controlled by a dedicated Router, or can be permanent with methods to show/hide the dialog. Depends on the use case.

**Reply by arkivanov** [reply]
Also please see the related discussion in the old repository: https://github.com/badoo/Decompose/discussions/59

**Reply by Kotlinlife** [reply]
Big thanks! I think toast message should be show same way with "callback" to state

**Comment by malliaridis** [comment]
Just for reference, because I tried to solve this issue with a dedicated router by using `ChildStack`. Besides `ChildStack` there is a better navigation element that can be used called `ChildOverlay` (see [here](https://arkivanov.github.io/Decompose/navigation/overlay/overview/#child-overlay-overview)).

I didn't know that until I tried to find a way to initialize a `ChildStack` without any child (and failed). 😅

**Reply by arkivanov** [reply]
Thanks for point that out! `ChildStack` can be also used to show a dialog, e.g. there could be two configurations - `None` and `Dialog`. And you could replace one with another. But indeed, `ChildOverlay` should be a better way for dialogs.

**Reply by malliaridis** [reply]
That's correct. However, with the configuration of `None` I still have to load a view into a container (at least the way I trie to set things up in Android with `StackRouterView`).

So `ChildOverlay` came for the rescue. 😄
